src/main/java/com/atlassian/db/replica/internal/ReplicaConnectionProvider.java

```diff
@@ -58,6 +58,9 @@ public int getTransactionIsolation() throws SQLException {
 
     public void setAutoCommit(Boolean autoCommit) throws SQLException {
         final boolean autoCommitBefore = getAutoCommit();
+        if (autoCommitBefore != autoCommit) {
```

Comment:
It's a draft because I didn't add `compatibleWithPreviousVersion`. I want to get the change reviewed first and then add `ifs`.

Reply:
What're your thoughts on adding those `ifs` every time in - **crucial place**, and boiler code in all top-down classes to provide if in the **crucial place**
I remember that at the beginning I predict that there was the conclusion to avoid any static providers.
I just want to ensure you still think so.

Reply:
> What're your thoughts on adding those ifs every time in - crucial place

I'm happy that we don't have to accumulate it, and we can clear it every release. In the product part, we accumulated 16 feature flags so far, while here we have only one. 
Yesterday I spend 3h pairing on the product flags dependencies and trying to unsolve the puzzle. I prefer to add the code (you just add the if and compiler + intellij will tell you where should you add the boilerplate). It's much easier than managing complex dependencies.

> every time in - crucial place, and boiler code in all top-down classes to provide if in the crucial place

We should not be afraid to change critical places. If there's a path uncovered by tests, we'll introduce the bug anyway in the future, and we add the missing coverage.

>  was the conclusion to avoid any static providers. I just want to ensure you still think so.

Yes. We should avoid static providers. Nothing changed here.